### ISSUE TITLE
doFactoryReset does not always preserve pairing information

### ISSUE THREAD

**Issue by korzonkiee** [polarofficial/polar-ble-sdk#519]
**Platform on which you observed the bug:** 
- [x] Android 
- [ ] iOS
- [ ] Other
- [ ] Platform is not relevant for this bug

**Device on which you observed the bug:**
- [ ] Polar OH1
- [ ] Polar Verity Sense
- [ ] Polar H10
- [ ] Polar H9
- [x] Polar 360
- [ ] Other
- [ ] Device is not relevant for this bug

**Describe the bug**
The `doFactoryReset` function is intended to reset the device while preserving pairing information, but it does not always behave as expected. Although the SDK passes true as the parameter for preservePairingInformation ([here](https://github.com/polarofficial/polar-ble-sdk/blob/master/sources/Android/android-communications/library/src/sdk/java/com/polar/sdk/impl/BDBleApiImpl.kt#L1992)), pairing information is sometimes lost, and as a result, user has to pair with the device again.

**How to Reproduce**
It doesn't always happen, but I experienced this behaviour 2-4 times in 10 attempts when doing factory reset on Android with Polar 360.

**Expected behavior**
Pairing information should be preserved during factory reset.

**Screenshots and logs**
[doFactoryResetPreservePairingInformation.log](https://github.com/user-attachments/files/18056970/doFactoryResetPreservePairingInformation.log)

**Comment by pth-pe-gh** [comment]
FLOW-61160

**Comment by TeemuLauPolar** [comment]
Hello from Polar! Thank you for contacting us. The issue you identified is not actually a bug, but a feature present in all Polar watches, including the Polar 360. Even if you perform a factory reset while preserving pairing information, the device will still track the number of resets. Devices with firmware versions earlier than 1.2.x will forget the pairing after five (5) factory resets or reboots. Devices with firmware versions 1.2.x or later will forget the pairing after ten (10) factory resets or reboots.

**Comment by orestesgaolin** [comment]
Thanks. Is the counter reset after the firmware update?

**Comment by TeemuLauPolar** [comment]
Yes, the counter is reset if the device can run for 60 seconds. Once the device has run for 60 seconds, it assumes it is stable and resets the counter to 0.
